### ISSUE TITLE
Icon window property is not honoured by X11 display

### ISSUE THREAD

**Issue by el-dee** [panda3d/panda3d#603]
Setting the runtime application icon using WindowProperties.setIconFilename() or usinr PRC property 'icon-filename' does not work on X11 display, the property is simply ignored (it works fine on osx display and win display).

To support this,  x11GraphicsWindow::set_wm_properties() should be modified to :
* get the allowed icon sizes from the WM using XGetIconSizes()
* create a pixmap from the icon bitmap using XCreateBitmapFromData() (and maybe resizing to the closest allowed icon size)
* set icon_pixmap field of wm_hints to the pixmap created above
* set IconPixmapHint in wm_hints flags

The tricky part being read the icon image and convert it to ARGB bitmap :)

**Comment by rdb** [comment]
X11GraphicsWindow already contains a `read_ico` function that reads an .ico file into an XCursor image; maybe we can leverage the same code to read icons?

**Comment by el-dee** [comment]
You're right and it's much easier. On Mac you can use almost any kind of image as icon, but that's the exception not the rule :)

**Comment by el-dee** [comment]
I have it working, I just need to clean up and make a proper PR. In the end I did not use XGetIconSizes() as modern WM does not force a given size for the icons and the returned list is empty. and when using NET_WM_ICON, the image is resized if needed instead of cropped/tilled with wm_hints.
